cli/src/commands/operation/undo.rs

```diff
@@ -75,6 +75,20 @@ pub fn cmd_op_undo(
         let template = tx.base_workspace_helper().operation_summary_template();
         template.format(&bad_op, formatter.as_mut())?;
         writeln!(formatter)?;
+
+        let was_undo_op = *tx.repo().view() == parent_op.view()?;
+        if args.operation == "@" && was_undo_op {
+            writeln!(
+                ui.hint_default(),
+                "This action reverted an 'undo' operation. The repository is now in the same \
+                 state as it was before the original 'undo'."
+            )?;
+            writeln!(
+                ui.hint_default(),
+                "If your goal is to undo multiple operations, consider using `jj op log` to see \
+                 past states, and `jj op restore` to restore one of these states."
+            )?;
+        }
```

Comment:
nit: I would move these hints after `tx.finish()`. Hints are often printed last. It's also better to not make both `ui.status_formatter()` and `ui.hint_default()` alive, while that wouldn't cause problems.